### PR TITLE
Completable.andThen(Completable)

### PR DESCRIPTION
**Pull request by CherryPerry**

### CODE REVIEW

File: reaktive-test/src/commonMain/kotlin/com/badoo/reaktive/test/completable/TestCompletableObserver.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    override fun onComplete() {
+        onEvent(Event.OnSuccess)
+    }
+
+    override fun onError(error: Throwable) {
+        onEvent(Event.OnError(error))
+    }
+
+    sealed class Event {
+        object OnSuccess : Event()
```
Comment (CherryPerry) [inline reaktive-test/src/commonMain/kotlin/com/badoo/reaktive/test/completable/TestCompletableObserver.kt]: Rename to `OnCompleted`.

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/AndThen.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+
+fun Completable.andThen(completable: Completable): Completable =
+    completableUnsafe { observer ->
+        val disposableWrapper = DisposableWrapper()
+        observer.onSubscribe(disposableWrapper)
+
+        subscribeSafe(
+            object : CompletableObserver, Observer, ErrorCallback by observer {
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/AndThen.kt]: `Observer` is useless and can be omitted, `CompletableObserver` interface extends `Observer`

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/AndThen.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onComplete() {
+                    completable.subscribeSafe(
+                        object : CompletableObserver by observer {
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/AndThen.kt]: Better to delegate only `CompletableCallbacks`

Comment (arkivanov) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/AndThen.kt]: Also you can add `Observer by this` to as you already have proper implementation of `Observer`.

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/AndThen.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+        subscribeSafe(
+            object : CompletableObserver, ErrorCallback by observer {
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onComplete() {
+                    completable.subscribeSafe(
+                        object : CompletableObserver, CompletableCallbacks by observer {
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/AndThen.kt]: Why can't we write:
```kotlin
                override fun onComplete() {
                    completable.subscribeSafe(
                        object : CompletableObserver, Observer by this, CompletableCallbacks by observer {
                        }
                    )
                }
```

**Pull request merged**